handler/datasource/inspect/itemdeals.go

```diff
@@ -0,0 +1,40 @@
+package inspect
+
+import (
+	"github.com/data-preservation-programs/singularity/model"
+	"gorm.io/gorm"
+)
+
+func GetItemDealsHandler(
+	db *gorm.DB,
+	id string,
+) ([]model.Deal, error) {
+	return getItemDealsHandler(db, id)
+}
+
+// @Summary Get all deals that have been made for an item
+// @Tags Data Source
+// @Accept json
+// @Produce json
+// @Param id path string true "Source ID"
+// @Success 200 {array} model.Deal
+// @Failure 500 {object} api.HTTPError
+// @Router /item/{id}/deals [get]
+func getItemDealsHandler(
+	db *gorm.DB,
+	id string,
+) ([]model.Deal, error) {
```

Comment:
You might want more information than this in case the item is splitted across different deals, i.e.
```
struct {
  model.Car
  []model.ItemPart
  []model.Deal
}
```
this tells which CAR file (piece_cid) contains which part of items and the related deals

Reply:
when you say split across multiple deals, do u mean multiple deals with the same ItemParts in them, or multiple deals with different ItemParts?